### PR TITLE
Small cleanup of summary uses

### PR DESCRIPTION
**Pull request by SkySkimmer**
- Remove Hints.add_hints_init
    
   It was only used in ltac/rewrite which as a plugin is too late to
   affect init.

- Replace trivial uses of declare_summary with summary.refs
    
   NB: The one in notation.ml looks trivial but is needed to do
   with_notation_protection (used by inductive/fixpoint local notations).

### CODE REVIEW

**Review by ppedrot — APPROVED** [review]

**Comment by SkySkimmer** [comment]
>Remove Hints.add_hints_init

>It was only used in ltac/rewrite which as a plugin is too late to
affect init.


Looking into this some more, it seems like something resurrected its usage after b594d3f6f2c5ae36dd1da727bb1707271b0b784d (probably a merge/rebase issue). That explains why things worked even when making ltac a plugin made it run too late to be useful.

File: library/nametab.ml [hunk 1]
```diff
@@ -279,10 +279,10 @@ module ExtRefTab = Make(FullPath)(ExtRefEqual)
 module MPTab = Make(FullPath)(MPEqual)
 
 type ccitab = ExtRefTab.t
-let the_ccitab = ref (ExtRefTab.empty : ccitab)
+let the_ccitab = Summary.ref ~name:"ccitab" (ExtRefTab.empty : ccitab)
```
Comment (ejgallego) [inline library/nametab.ml]: Very minor nit, maybe a structured naming `tab:cci` would be nicer. Understanding the summary is essential in work with workers so every bit helps.

**Review by ejgallego — APPROVED** [review]
Nice cleanup.

**Comment by ppedrot** [comment]
It seems that this needs a rebase.

**Comment by ppedrot** [comment]
@mattam82 @silene Last round to complain. That's simple enough that if I don't get news from you guys I'll merge that soon.

**Pull request merged**